enhancements/authentication/direct-external-oidc-provider.md

```diff
@@ -212,6 +212,83 @@ These changes have already been implemented, and the initial PR for them can be
+
+OpenShift's default admission plugin `authorization.openshift.io/RestrictSubjectBindings`
+utilizes the OAuth stack, specifically the `User` and `Group` APIs, to ensure `RoleBinding`
+subjects do not violate any restrictions in the namespace they are going to be created in.
+These restrictions are specified by `RoleBindingRestriction` custom resources.
+
+Removing the OAuth stack entirely during external OIDC setup may result in the `authorization.openshift.io/RestrictSubjectBindings` admission plugin not functioning as expected.
+
+Anticipated impacts are:
+
+- Logs in the KAS reflecting that informers for OAuth APIs could not be started
+- Logs in the KAS reflecting that the oauth-apiserver is unreachable
+- Errors, that are not actionable from the perspective of users, occurring during validation of `RoleBinding` resources when a `RoleBindingRestriction` resource in the same namespace specifies user/group restrictions
+
+These impacts may cause concern for cluster administrators that their cluster is in an unhealthy state and may not be easily correlated to the enablement of the OIDC functionality.
+
+To ensure a holistic change in cluster behavior when enabling the OIDC functionality, and reduce potentially concerning behavior in the control plane components, it is proposed that changes are made to:
+
+- Disable the `authorization.openshift.io/RestrictSubjectBindings` and `authorization.openshift.io/ValidateRoleBindingRestriction` admission plugins
+- Remove the `rolebindingrestrictions.authorization.openshift.io` CustomResourceDefinition
+- Disable the OpenShift oauth-apiserver availability check
+
+##### Changes to the kube-apiserver
+
+The OpenShift-specific patch to the kube-apiserver that adds this admission plugin is found here:
+https://github.com/openshift/kubernetes/blob/3c62f738ce74a624d46b4f73f25d6c15b3a80a2b/openshift-kube-apiserver/openshiftkubeapiserver/patch.go#L71
+
+In order to prevent misleading logs about informers that failed to start or failure to connect to the oauth-apiserver, the following changes to this patch are to be made:
+
+- Add a new default flag, `--openshift-oauth-not-desired`, that gates the oauth-apiserver availability check. The availability check will be run when `--openshift-oauth-not-desired` is not set and not run when it is set.
+- Informers for the `Group` API are only configured and started as part of the first run of the `authorization.openshift.io/RestrictSubjectBindings` admission plugin validation loop. This makes it such that the informer will not be configured or attempt to start when the admission plugin is disabled.
+
+##### Changes to the cluster-kube-apiserver-operator
+
+When the `Authentication` resource with name `cluster` has `.spec.type` set to `OIDC`, the `authorization.openshift.io/RestrictSubjectBindings` and `authorization.openshift.io/ValidateRoleBindingRestriction` admission plugins will be disabled.
```

Comment:
Seems like the admission plugin should be disabled when .spec.type is set to something other than `IntegratedOAuth` since the user/group mappings will be invalid.

Reply:
Ah, this is something @liouk and I had an open question on and were going to ask in the arch call tomorrow. We weren't sure if setting `None` was equivalent to "I don't want the OpenShift oauth server" and should result in this teardown behavior or if it was more of a "keep the oauth server up and configured, but don't manage it anymore".

I think it sounds reasonable to have this be the behavior when anything other than `IntegratedOAuth` is set.